app/models/view/inventory.rb

```diff
@@ -82,7 +82,7 @@ def quantity_for(storage_location: nil, item_id: nil)
 
       if storage_location
         if item_id
-          @inventory.storage_locations[storage_location.to_i].items[item_id.to_i]&.quantity || 0
+          @inventory.storage_locations[storage_location.to_i]&.items&.dig(item_id.to_i)&.quantity || 0
```

Comment:
Unrelatedly, I learned a while back that javascript's `?` operator is short-cutting, so that the first one in a long chain immediately halts the whole sequence. I like the explicitness of Ruby's a bit better, though it ends up more verbose.